Refactoring to tslint

BCybercell opened the pull request:
There are a few issues with regards to an entire file being overwritten, please advise.

BCybercell committed:
Refactored controllers

BCybercell committed:
Refactored entities

BCybercell committed:
Refactored providers

BCybercell committed:
Refactored services

BCybercell committed:
Refactored Guards

BCybercell committed:
Refactored members

BCybercell committed:
Refactored login, services and app

EvansMatthew97 commented on server/src/providers/shortest-path.service.spec.ts:
```diff
@@ -102,76 +102,77 @@ const line = [[
-    describe('Tests to see if there is more than 0 dams on the map',()=>{
-        it('More than 0 dams are on the map',async()=>{
+// -check features received from openView maps (roads > 0, dams > 0)
+    const Correcttop = - 25.8415;
```
These should probable be lowercase camelCase

EvansMatthew97 commented on server/src/providers/shortest-path.service.ts:
```diff
@@ -13,37 +13,41 @@ export class ShortestPathService {
+    jp1 = this.index(j + 1),
+    im1 = this.index(i - 1),
+    ip1 = this.index(i + 1);
+  let s =
```
Should probably be on same line as next line

EvansMatthew97 commented on server/src/services/animal-interest-point.service.ts:
```diff
@@ -7,27 +7,23 @@ export class AnimalInterestPointService {
-            animalIntrestPoints.longatude = "28.282984";
-            animalIntrestPoints.latitude = "-25.865828";
-            return data.manager.save(animalIntrestPoints).then(animalIntrestPoints => { console.log("Saved a new animal intrestpoint with id: " + animalIntrestPoints.id) });
+        const addAnimal = con.then(async (data) => {
```
This seems messy.

Maybe something along the lines of 
```
const con = await this.databaseService.getConnection();
const animalInterestPoint = new AnimalInterestPoint();
...
const addedAnimal = await con.manager.save(animalInterestPoint);
```
would be better (get rid of .then and make singular instead of plural)?

BCybercell replied:
addAnimalInterestPoint(): boolean {
        const con = await this.databaseService.getConnection();
        const animalInterestPoints = new AnimalInterestPoint();

        animalInterestPoints.name = 'saltLick';
        animalInterestPoints.pointDescription = 'Saltlick at dam point A';
        animalInterestPoints.longitude = '28.282984';
        animalInterestPoints.latitude = '-25.865828';
            // tslint:disable-next-line:no-console
        const addedAnimal = await con.manager.save(animalInterestPoints);
        console.log('Saved a new animal interest point with id: ' + animalInterestPoints.id);
        return addedAnimal != null;
    }


like this? 
also it says I shoulf make this function async?

EvansMatthew97 commented on server/src/services/user.service.ts:
```diff
@@ -1,137 +1,113 @@
+            if (ExistingUser.token === aToken) {
                // console.log("User token in database" + ExistingUser.token);
-               var temp = new Date();
+               const temp = new Date();
```
Call this variable something more useful than "temp". Maybe "now"?

EvansMatthew97 reviewed (changes requested):
Minor changes

Looks good otherwise 👍 

Make sure that all the files have a line break at the end if you haven't yet, please.

BCybercell committed:
Refactored requested files
Added camel case
Changed async call to await
Changed temp variable to now, more specific
